### ISSUE TITLE
[Windows/Wasm] Default selected items from ItemsRepeaterExtensions multiselection are unselectable on the first click

### ISSUE THREAD

**Issue by rajamatt** [unoplatform/uno.toolkit.ui#1194]
## Current behavior

Setup: ItemsRepeater with utu:ItemsRepeaterExtensions.SelectionMode="Multiple" with items selected by default.
Problem: Makes selected items that are selected by default when loading the page unselectable on the first click.

This sample is using a ListViewItem as the ItemsRepeater's ContentTemplate, but it also happens with other selection controls such as ToggleButton.

Here, the item "two" is selected by default when opening the app:
![ListViewItemsRepeater](https://github.com/unoplatform/uno.extensions/assets/70542961/4230defa-39b8-4a7f-9a0c-e84743e9e44f)

Minimal repro app: [ItemsRepeaterSelection.zip](https://github.com/user-attachments/files/15917885/ItemsRepeaterSelection.zip)

## Environment

Nuget Package (s):

Package Version(s):

Affected platform(s):

- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Android
- [ ] WebAssembly
- [ ] Windows
- [ ] Skia (WPF)
- [ ] Skia (GTK on Linux/macOS/Windows)
- [ ] Skia (Linux Framebuffer)
- [ ] Build tasks

Visual Studio:

- [ ] 2019 (version: )
- [ ] 2022 (version: )
- [ ] Visual Studio Code (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by kazo0** [comment]
@Kunal22shah I believe this is the same root issue that we were discussing with the itemsrepeater, I will link this as the blocker for that issue in the backlog

**Comment by dr1rrb** [comment]
Could be either the Toolkit's ItemsRepeaterExtensions which does not restore the initial/current selection when the collection is data-bound, either MVUX which pushes that initial selection with a delay (as `ISelectionInfo` does not provided any way to update teh selection from the source to the view, the selection has to be sync)

**Comment by Xiaoy312** [comment]
the data came with desync'd states:
![image](https://github.com/user-attachments/assets/fc53f885-818a-4c62-9311-7c49a95022b9)
^ GetSelectedRanges just returns empty array of ItemIndexRange

you can even repro it without toolkit:
```cs
public MainPage()
{
	this.InitializeComponent();
	SUT.RegisterPropertyChangedCallback(ItemsRepeater.ItemsSourceProperty, IR_ItemsSourceChanged);
	this.DataContext = new BindableMainModel();
}
private void IR_ItemsSourceChanged(DependencyObject sender, DependencyProperty dp)
{
	if (sender is not ItemsRepeater ir) return;
	if (ir.ItemsSource is ISelectionInfo si)
	{
			var ranges = si.GetSelectedRanges(); // new ItemIndexRange[0]
			var rangesDebug = string.Join(", ", ranges.Select(x => x.Length > 1 ? $"[{x.FirstIndex}-{x.LastIndex}]" : $"[{x.FirstIndex}]"));

			var isSelecteds = Enumerable.Range(0, (si as BindableListFeed<string>)?.Count ?? 0)
				.Select(si.IsSelected)
				.ToArray(); // false, true, false
	}
}

**Comment by dr1rrb** [comment]
@Xiaoy312  Is this on windows?

**Comment by kazo0** [comment]
@rajamatt I see there is a draft PR for this that is Windows-specific. This issue is also occurring on WASM no?

**Comment by rajamatt** [comment]
@kazo0 Yes the Wasm problem is in toolkit, I didn't get the chance to finalize the Wasm changes yet

**Comment by kazo0** [comment]
> Wasm problem is in toolk

@rajamatt whats the issue in toolkit? do we have a separate issue for that? We can split these issues if they are different problems

**Comment by rajamatt** [comment]
@kazo0 Yeah both repos produce the same bug but for different reasons, on Windows in uno.extensions it's because we weren't setting the ``_localSelection`` when constructing. On Wasm in uno.toolkit.ui it's because there's some delay, the ItemsRepeater.ItemsSource is populated with empty data and then populated after a certain delay with the real data. Binding an event to ItemsRepeater.ItemsSourceView.CollectionChanged fixes this (I wish I had some code to show but my branch is a bit broken right now)

**Comment by kazo0** [comment]
@rajamatt yes let's create a separate issue and we can edit this one for the Windows specific scenario. Reach out to @Xiaoy312 and I if you need input on the fix for the ItemsRepeater one
